doc/userguide/src/CreatingTestData/AdvancedFeatures.rst

```diff
@@ -397,7 +397,104 @@ lower limit, upper limit and step.
    Float parameters  [Documentation]  Loops over   values     3.14, 4.34,  and 5.34
    \                 :FOR             ${index}     IN RANGE   3.14         6.09      1.2
    \                                  Log          ${index}
-   ================  ===============  ===========  =========  ===========  ========  =======
+   ================  ===============  ===========  =========  ===========  ========  ========
+
+For in enumerate
+~~~~~~~~~~~~~~~~
+
+Sometimes it is useful to loop over a list and also keep track of your location
+inside the list.  Robot Framework has a special
+`FOR index ... IN ENUMERATE ...` syntax for this situation.
+This syntax is derived from the
+`Python built-in function <https://docs.python.org/2/library/functions.html#enumerate>`_.
+
+:name:`:FOR IN ENUMERATE` loops work just like regular :name:`:FOR` loops,
+except the cell after its loop variables must say :name:`IN ENUMERATE`,
+and they must have an additional index variable before any other loop-variables.
+That index variable has a value of `0` for the first loop, `1` for the second, etc.
+
+For example, the following two test cases do the same thing:
+
+
+.. table:: For in enumerate comparison
+   :class: example
+   =================  ================  ===============  =============  ===================  ========
```

Comment:
There should be an empty row after the table declaration and the table itself:

```restructuredtext
.. table:: For in enumerate comparison
   :class: example

=================  ================  ===============  =============  ===================  ========
```

After adding empty row to all examples I was able to generate User Guide without problems.